packages/web/src/common/store/pages/saved/lineups/sagas.js

```diff
@@ -178,20 +211,30 @@ function* watchRemoveFromLibrary() {
             : LibraryCategory.Repost
       })
     )
-    if (removedTrackUid) {
-      yield put(savedTracksActions.remove(Kind.TRACKS, removedTrackUid))
-      if (
-        removedTrackUid !== playerUid &&
-        queueSource === QueueSource.SAVED_TRACKS
-      ) {
-        yield put(queueActions.remove({ uid: removedTrackUid }))
+
+    if (
+      (type === UNSAVE_TRACK && currentCategory === LibraryCategory.Favorite) ||
+      (type === UNDO_REPOST_TRACK && currentCategory === LibraryCategory.Repost)
+    ) {
+      const playerUid = yield select(getPlayerUid)
+      const queueSource = yield select(getSource)
+      if (removedTrackUid) {
+        yield put(savedTracksActions.remove(Kind.TRACKS, removedTrackUid))
+        if (
+          removedTrackUid !== playerUid &&
+          queueSource === QueueSource.SAVED_TRACKS
+        ) {
+          yield put(queueActions.remove({ uid: removedTrackUid }))
+        }
       }
-    }
-    const lineupSaveUid = yield select(getSavedTracksLineupUid, { id: trackId })
-    if (lineupSaveUid) {
-      yield put(savedTracksActions.remove(Kind.TRACKS, lineupSaveUid))
-      if (lineupSaveUid !== playerUid) {
-        yield put(queueActions.remove({ uid: lineupSaveUid }))
+      const lineupSaveUid = yield select(getSavedTracksLineupUid, {
```

Comment:
Not a code comment, but just for my own understanding  - I think I grok'ed what the code above this does, let me see if I'm right:
1. Get the UID of the track that was removed from the library (either by unfavoriting or unreposting)
2. If the currently viewed category matches the action, remove that track from the page's table, and if the track is not the one currently playing, remove it from the queue

But this code looks to do that too so I'm confused - what's the difference between `removedTrackUid` and `lineupSaveUid`?

Reply:
omg i think i just moved things around poorly and ended up duplicating a lot of this. will fix, tyty! `removedTrackUid` and `lineupSaveUid` should be the same lol

Reply:
Ok actually this code was like this before and I just moved it into the new `if`. 
If the track was added in the current session, its uid would be in `removedTrackUid` which used to be named `localSaveUid` (I'll change it back since that was a better name). Otherwise it would be in `lineupSaveUid`

Reply:
Ahh I see, so it's the local vs server thing! Thanks, appreciate you taking the time to explain!